plasmapy/particles/particle_class.py

```diff
@@ -2023,7 +2022,88 @@ def mass(self, m: u.kg):
                 ) from exc
 
 
-# TODO: Describe valid particle representations in docstring of particle_like
-
-particle_like = Union[str, Integral, Particle]
-"""A typing construct for valid representations of a particle."""
+particle_like = Union[str, Integral, Particle, CustomParticle]
+
+particle_like.__doc__ = """
+An `object` is particle-like if it can be identified as an instance of
+`~plasmapy.particles.particle_class.Particle` or
+`~plasmapy.particles.particle_class.CustomParticle`, or cast into one.
```

Comment:
At the moment, `particle_like` is not an instance of `GenericAlias`...though I suspect that is the way we should go about doing this in the future.  Good idea to point out that it's for typing annotations.

Reply:
I'll admit that I don't fully understand the mechanics/lexicon behind python's `typing` and `types` pkgs, but this...

<img width="430" alt="image" src="https://user-images.githubusercontent.com/29869348/107155701-b15ce780-692e-11eb-9f09-a743d77e30c1.png">

suggests to me that `particle_like` is a Generic Alias.

Reply:
But I really don't care if `GenericAlias` is mentioned at all.  I feel most people will probably not know what that means anyways.  ...I mean, I've read about it and I still don't fully understand it... As you said above, the main point is to say somewhere in the documentation that this variable is intended for typing annotations.

Reply:
Yeah...the details of this are not particularly clear...
```Python
>>> typing_alias = typing.Union[int, set]
>>> isinstance(typing_alias, typing.GenericAlias)
False
```
> As you said above, the main point is to say somewhere in the documentation that this variable is intended for typing annotations.

Agreed completely!  In part because that reduces the need for me to actually understand things...

Reply:
Honestly, reading through the `typing` document I get the feeling that the pkg is going through some kind change. Some of the document feels a little out of date and some feels like it's been recently modified.  Look at the Generic Alias section, the link for `GenericAlias` (the class) is nonexistent or broken.